rasa/core/policies/unexpected_intent_policy.py

```diff
@@ -0,0 +1,914 @@
+        to or above the threshold.
+
+        Args:
+            label_quantiles: Quantiles computed for each label id
+            tolerance: Specified tolerance value from the configuration.
+
+        Returns:
+            Computed thresholds
+        """
+        label_thresholds = {}
+        for label_id in label_quantiles:
+            num_thresholds = len(label_quantiles[label_id])
+            label_thresholds[label_id] = label_quantiles[label_id][
+                min(int(tolerance * num_thresholds), num_thresholds - 1)
+            ]
+        return label_thresholds
+
+    def persist_model_utilities(self, model_path: Path) -> None:
+        """Persists model's utility attributes like model weights, etc.
+
+        Args:
+            model_path: Path where model is to be persisted
+        """
+        super().persist_model_utilities(model_path)
+        io_utils.pickle_dump(
+            model_path / f"{self._metadata_filename()}.label_quantiles.pkl",
+            self.label_quantiles,
+        )
+
+    @classmethod
+    def _load_model_utilities(cls, model_path: Path) -> Dict[Text, Any]:
+        """Loads model's utility attributes.
+
+        Args:
+            model_path: Path where model is to be persisted.
+        """
+        model_utilties = super()._load_model_utilities(model_path)
+        label_quantiles = io_utils.pickle_load(
+            model_path / f"{cls._metadata_filename()}.label_quantiles.pkl"
+        )
+        model_utilties.update({"label_quantiles": label_quantiles})
+        return model_utilties
+
+    @classmethod
+    def _update_loaded_params(cls, meta: Dict[Text, Any]) -> Dict[Text, Any]:
+        meta = train_utils.override_defaults(cls.defaults, meta)
+        return meta
+
+    @classmethod
+    def _load_policy_with_model(
+        cls,
+        model: "IntentTED",
+        featurizer: TrackerFeaturizer,
+        model_utilities: Dict[Text, Any],
+        should_finetune: bool,
+    ) -> "UnexpecTEDIntentPolicy":
+        return cls(
+            featurizer=featurizer,
+            priority=model_utilities["priority"],
+            model=model,
+            fake_features=model_utilities["fake_features"],
+            entity_tag_specs=model_utilities["entity_tag_specs"],
+            should_finetune=should_finetune,
+            label_quantiles=model_utilities["label_quantiles"],
+            **model_utilities["meta"],
+        )
+
+
+class IntentTED(TED):
+    """Follows TED's model architecture from https://arxiv.org/abs/1910.00486.
+
+    However, it has been re-purposed to predict multiple
+    labels (intents) instead of a single label (action).
+    """
+
+    def _prepare_label_classification_layers(self, predictor_attribute: Text) -> None:
+        """Prepares layers & loss for the final label prediction step."""
+        self._prepare_embed_layers(predictor_attribute)
+        self._prepare_embed_layers(LABEL)
+        self._prepare_dot_product_loss(LABEL, self.config[SCALE_LOSS])
```

Comment:
As far as I can see, this is an exact copy of what's in `TransformerRasaModel` in `models.py`. I don't think it's needed here. TED doesn't override it, so we should be able to use the original version from `TransformerRasaModel` in a child class like `IntentTED`.

Reply:
Good catch! Removing it now.